scio-extra/src/main/scala/com/spotify/scio/extra/sparkey/ShardedSparkeyUri.scala

```diff
@@ -84,11 +83,11 @@ trait ShardedSparkeyUri extends SparkeyUri {
 }
 
 private[sparkey] object ShardedSparkeyUri {
-  def apply(basePath: String, options: PipelineOptions): ShardedSparkeyUri =
+  def apply(basePath: String, rfuCreateFn: () => RemoteFileUtil): ShardedSparkeyUri =
```

Comment:
Why do we need a `fn` here? Can't seem to find a place where it's not `RemoteFileUtil.create()`.

Reply:
The original code deferred creation of `RemoteFileUtil` until it knew it was dealing with a remote uri, so this is replicating that behavior. We could also eagerly create `RemoteFileUtil` but it would be unused when reading locally.